cmd/status-cli/message.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/requests"
+
+	"github.com/urfave/cli/v2"
+)
+
+func sendContactRequest(cCtx *cli.Context, from *StatusCLI, toID string) error {
+	from.logger.Info("send contact request, contact public key: ", toID)
+	request := &requests.SendContactRequest{
+		ID:      toID,
+		Message: "Hello!",
+	}
+	resp, err := from.messenger.SendContactRequest(cCtx.Context, request)
+	from.logger.Info("function SendContactRequest response.messages: ", resp.Messages())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func sendContactRequestAcceptance(cCtx *cli.Context, from *StatusCLI, msgID string) error {
+	from.logger.Info("accept contact request, message ID: ", msgID)
+	resp, err := from.messenger.AcceptContactRequest(cCtx.Context, &requests.AcceptContactRequest{ID: types.Hex2Bytes(msgID)})
+	if err != nil {
+		return err
+	}
+	from.logger.Info("function AcceptContactRequest response: ", resp.Messages())
+
+	return nil
+}
+
+func sendDirectMessage(ctx context.Context, from *StatusCLI, text string) error {
+	if len(from.messenger.MutualContacts()) == 0 {
+		return nil
+	}
+	chat := from.messenger.Chat(from.messenger.MutualContacts()[0].ID)
+	from.logger.Info("send message to contact: ", chat.ID)
+
+	clock, timestamp := chat.NextClockAndTimestamp(from.messenger.GetTransport())
+	inputMessage := common.NewMessage()
+	inputMessage.ChatId = chat.ID
+	inputMessage.LocalChatID = chat.ID
+	inputMessage.Clock = clock
+	inputMessage.Timestamp = timestamp
+	inputMessage.MessageType = protobuf.MessageType_ONE_TO_ONE
+	inputMessage.ContentType = protobuf.ChatMessage_TEXT_PLAIN
+	inputMessage.Text = text
+
+	_, err := from.messenger.SendChatMessage(ctx, inputMessage)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func retrieveMessagesLoop(ctx context.Context, cli *StatusCLI, tick time.Duration, msgCh chan string, wg *sync.WaitGroup) {
```

Comment:
@Passing `wg *sync.WaitGroup` seems wrong to me. Why should a function know how it will be launched?
You should either wrap like this in the place of calling:
```go
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
          defer wg.Done()
          retrieveMessagesLoop(ctx, messenger, RetrieveInterval, msgCh)
    }()
```

or create a separate `StartRetrieveMessagesLoop` function which will do this.